core/src/main/java/com/datastax/oss/driver/internal/core/type/codec/VectorCodec.java

```
@@ -156,4 +116,195 @@ public CqlVector<SubtypeT> parse(@Nullable String value) {
         : CqlVector.from(value, this.subtypeCodec);
   }
+
+  private static class FixedLength<SubtypeT> implements VectorCodecProxy<SubtypeT> {
+    private final VectorType cqlType;
+    private final TypeCodec<SubtypeT> subtypeCodec;
+
+    private FixedLength(VectorType cqlType, TypeCodec<SubtypeT> subtypeCodec) {
+      this.cqlType = cqlType;
+      this.subtypeCodec = subtypeCodec;
+    }
+
+    @Override
+    public ByteBuffer encode(
+        @Nullable CqlVector<SubtypeT> value, @NonNull ProtocolVersion protocolVersion) {
+      if (value == null || cqlType.getDimensions() <= 0) {
+        return null;
+      }
+      ByteBuffer[] valueBuffs = new ByteBuffer[cqlType.getDimensions()];
+      Iterator<SubtypeT> values = value.iterator();
+      int allValueBuffsSize = 0;
+      for (int i = 0; i < cqlType.getDimensions(); ++i) {
+        ByteBuffer valueBuff;
+        SubtypeT valueObj;
+
+        try {
+          valueObj = values.next();
+        } catch (NoSuchElementException nsee) {
+          throw new IllegalArgumentException(
+              String.format(
+                  "Not enough elements; must provide elements for %d dimensions",
+                  cqlType.getDimensions()));
+        }
+
+        try {
+          valueBuff = this.subtypeCodec.encode(valueObj, protocolVersion);
+        } catch (ClassCastException e) {
+          throw new IllegalArgumentException("Invalid type for element: " + valueObj.getClass());
+        }
+        if (valueBuff == null) {
+          throw new NullPointerException("Vector elements cannot encode to CQL NULL");
+        }
+        allValueBuffsSize += valueBuff.limit();
+        valueBuff.rewind();
+        valueBuffs[i] = valueBuff;
+      }
+      // if too many elements, throw
+      if (values.hasNext()) {
+        throw new IllegalArgumentException(
+            String.format(
+                "Too many elements; must provide elements for %d dimensions",
+                cqlType.getDimensions()));
```

Just for my own edification: we did decide we were going to throw in this case, right?  The other option is that we can silently ignore any extra data that might be present in the buffer and I recall some conversation on this point.  My recollection was that we had decided _not_ to throw in this case but my PR for PYTHON-1369 explicitly [does throw](https://github.com/datastax/python-driver/pull/1217/files#diff-18308cd34ffea992ea325d7d0c09eb78b27203def63c60dedbbcd8db8626cd7fR486-R489) in this case as well.

I do remember that we had decided to keep this behaviour consistent across drivers.  @joao-r-reis can you confirm what the C# driver does in this case?

---

The C# driver throws. Silently ignoring seems like a bad idea imo

---

@SiyaoIsHiding hit me up with a link to a Slack conversation in which a decision was announced on this point.  The decision sprang out of a conversation the poster had with @joao-r-reis about our desire to standardize behaviour on this point across the drivers and how the default behaviour should be to throw.  The poster also pointed out that they'd already made changes to the Python driver to bring about this change in behaviour.

That poster was me.

🤦 

Move along, nothing to see here...